lib/isafw/isaplugins/ISA_cve_plugin.py

```diff
@@ -161,17 +171,21 @@ def process_report_type(self, rtype):
             flog.write("Args: " + args)
         try:
             popen = subprocess.Popen(
-                args, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-            stdout_value = popen.communicate()[0]
+                args, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            result = popen.communicate()
         except:
-            stdout_value = "Error in executing cve-check-tool"
+            tool_stderr_value = "Error in executing cve-check-tool" + str(sys.exc_info())
             with open(self.logfile, 'a') as flog:
                 flog.write("Error in executing cve-check-tool: " +
-                           sys.exc_info())
+                           str(sys.exc_info()))
         else:
-            report = self.report_name + "." + rtype
-            with open(report, 'wb') as freport:
-                freport.write(stdout_value)
+            stdout_value = result[0]
+            tool_stderr_value = result[1]
+            if not tool_stderr_value :
```

Comment:
The assumption here is that there is an empty stderr if and only if there was no problem.

But that assumption is not correct: for example, if the process gets killed prematurely, it will exit with a non-zero exit code but with no output via stderr.

What you need to add is a check of popen.returncode == 0 and then add some explanation of the non-zero exit code to the tool_stderr_value (something like "[Command terminated with non-zero exit code %d.]" which cannot be confused with output by the tool itself). That way it is never empty in case of a failure.